src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -80,12 +81,35 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
-    float sentimentScore = getSentimentScore(text);
-    
-    Message message = new Message(user, text, sentimentScore);
-    datastore.storeMessage(message);
 
+    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+
+    String regex = "(https?://\\S+\\.(png|jpg|gif))";
+
+    String replacement = "<img src=\"$1\" />";
+
+
+    String youtube_regex = "(https://www.youtube.com/watch\\?v=(\\S*))";
+    String youtube_replacement = "<iframe width=\"560\" height=\"315\" "+
+   "src=\"https://www.youtube.com/embed/$2\" frameborder=\"0\" "+
+   "allow=\"accelerometer; autoplay; encrypted-media; gyroscope; "+
+   "picture-in-picture\" allowfullscreen></iframe>";
+
+    String result = userText.replaceAll(youtube_regex, youtube_replacement);
+    String textWithImagesReplaced = userText.replaceAll(regex, replacement);
+    float sentimentScore;
+
+    if(userText.equals(result)){
+      sentimentScore = getSentimentScore(textWithImagesReplaced);
+      Message message = new Message(user, textWithImagesReplaced, sentimentScore);
+      datastore.storeMessage(message);
+    }
+    else{
+      sentimentScore= getSentimentScore(result);
+      Message message = new Message(user, result, sentimentScore);
+      datastore.storeMessage(message);
+    }
```

Comment:
I think this whole thing could be condensed a lot. 

Note that the only thing different is the value of the text variable. The sentiment score calculation changes but the whole thing doesn't need to be repeated (also, it doesn't make much sense to do this but then calculate the sentiment using the regex'd text?). I also think the `userText.equals(result)` is dubious because it's just for YouTube videos. I would rewrite this whole part, it's hard to read but I'm also concerned it's introducing misbehavior.

Reply:
I understand what you mean about the condensing part. I can make that wayy shorter. as far as the usertext.equals(result), it basically prevents from 2 messages being outputted - one message with the image and the second with the url. I am not really understanding what you mean about the misbehavior part bc its functional on the user page but not on the feed. @dj-smith

Reply:
Yeah I see now where the confusion is. And the variables have to stay separate because you only want to replace one link with a div, and not have HTML text in the message if a link isn't the one that gets replaced. Also, what's the expected behavior if there are multiple image links in a message?

I'd probably refactor this by just checking once if the text has changed, then assigning the parameter based on that check. Like, `boolean textChanged = !userText.equals(result)`. Or,

`userText = (!userText.equals(result)) ? textWithImagesReplaced : result;`

and then

`sentimentScore= getSentimentScore(userText);
Message message = new Message(user, userText, sentimentScore);
datastore.storeMessage(message);`

Reply:
So, to clarify, do we want to post videos and picture in the same post? If, so we may want to change the if/else branch to two separate if statements.

Reply:
Right, I'm also wondering what our desired behavior for that situation is. I think we should?

Reply:
We'd also need to change the replacement behavior so that the same text is processed by both regexes, rather than one or the other into two separate variables.

Reply:
yup fixed that!

Reply:
Can you elaborate as to what you mean about the replacement behavior @dj-smith . I was looking at the code and i thought that we were using youtube, img,and gifs to upload images and vids at this moment so I just made two replacement variables, one using an img tag and the other using an iframe tag.

Reply:
Well, as far as I can tell, right now (in this PR at least, I don't know if this is fixed in the other one) you take the original text and sort of process it in two "branches": you replace the youtube links and store them in `result`, and you replace the images and store them in `textWithImagesReplaced`.

Then, using that if statement, you _either_ set the text of the message to the image version _or_ the youtube version.

The effect would be (or at least this is how it seems to me) that you can't possibly replace both a video and an image link in the same message, because the text is processed in that branchlike way.